base/views.py

```diff
@@ -1,22 +1,42 @@
-from django.shortcuts import render, redirect
+from django.shortcuts import render, redirect , get_object_or_404
 from django.http import HttpResponse
 from django.contrib.auth.decorators import login_required
 from .models import User
-from django.contrib.auth import get_user_model
 from django.contrib.auth.forms import AuthenticationForm
-from django.contrib.auth import authenticate, login , logout
+from django.contrib.auth import authenticate, login , logout , get_user_model
 from django.contrib import messages
 from django.db import IntegrityError
 from datetime import date, timedelta
+from django.urls import reverse
+from django.core.mail import send_mail
+from django.utils.crypto import get_random_string
+from django.conf import settings
 
 # Create your views here.
 User = get_user_model()
 
+
+def send_email(request):    
+    return redirect('login')
+
+
 def home(request):
     return render(request, 'main.html')
 @login_required(login_url="/login")
 def dashboard(request):
     return render(request, 'base/dashboard.html')
+
+def verify_email(request, token):
+    user = get_object_or_404(User, verification_token=token)
+
+    if not user.is_verified:
+        user.is_verified = True
+        user.save()
+        messages.success(request, 'Your email has been verified successfully. You can now log in.')
+        return render(request, 'base/verification_success.html')
+    else:
+        messages.info(request, 'Your email is already verified.')
+        return render(request, 'base/verification_success.html')
```

Comment:
if user is already verified, redirect him to login page, with a message, you are already verified